### PR TITLE
API Platform upgrade: Removal of Deprecations

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 2.0                   |
| Bug fix?        | no                                                      |
| New feature?    | no                                                      |
| BC breaks?      | yes                                                       |
| Deprecations?   | no  |
| License         | MIT                                                          |

This PR aims to lay the groundwork for API Platform upgrade by removing obsolete API Platform FQCNs and making minor code improvements.

Second, it restores specification testing for classes that do not require any changes

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ChangePaymentMethodHandler.php
Comment (Rafikooo) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ChangePaymentMethodHandler.php:1]: Changes regarding `MessageHandlerInterface` will be extracted to a separate PR

**Review by Wojdylak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/RestrictingFilterEagerLoadingExtension.php [hunk 1]
```diff
@@ -13,32 +13,41 @@
...
-use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\Operation;
 use Doctrine\ORM\QueryBuilder;
 
 /**
  * @experimental
  * This class decorates api_platform.doctrine.orm.query_extension.filter_eager_loading.
  * It is a workaround for https://github.com/api-platform/core/issues/2253.
```
Comment (TheMilek) [inline src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/RestrictingFilterEagerLoadingExtension.php:24]: Just wondering if we should check if its still valid after bump 🤔

File: src/Sylius/Bundle/ApiBundle/Resources/config/services.xml [hunk 2]
```diff
@@ -15,36 +15,45 @@
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd"
 >
-<!--    <imports>-->
-<!--        <import resource="services/**/*.xml" />-->
```
Comment (TheMilek) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services.xml:19]: I think we should leave this line commented as we should go back to importing everything in one line to not forget about it

**Review by TheMilek — APPROVED** [review]

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/RestrictingFilterEagerLoadingExtension.php (hunk 1 above)
Comment (Rafikooo) [reply src/Sylius/Bundle/ApiBundle/Doctrine/QueryCollectionExtension/RestrictingFilterEagerLoadingExtension.php:24]: Good catch, but I think we should face this when converting a resource using this service

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -17,6 +17,17 @@
 >
     <imports>
 <!--        <import resource="services/**/*.xml" />-->
+        <import resource="services/api_payment_methods.xml" />
+        <import resource="services/applicators.xml" />
+        <import resource="services/context_builders.xml" />
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services.xml:22]: @GSadee `services/context_builders.xml` imported twice

Comment (diimpp) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services.xml:22]: Sorry, wrong @mention -- @Rafikooo

Comment (TheMilek) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services.xml:22]: @diimpp removed this duplication [here](https://github.com/Sylius/Sylius/pull/15393/commits/f1326ae1f9fe74372903972bfd2193e1ba575152#diff-32d98f7d054993ed0efa9e1348323ca84ff552c710dfa488b42fd6de6d494a1fL22)

Comment (Rafikooo) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services.xml:22]: Thanks 💃
